scripts/irods/test/test_misc.py

```diff
@@ -512,3 +508,31 @@ def test_configuration_validation_reports_fqdn_size_violations__issue_8287(self)
                 self.assertIn('"valid": false', res.stderr)
                 self.assertIn('"instanceLocation": "/host"', res.stderr)
                 self.assertIn('"error": "Number of characters must be at most 253"', res.stderr)
+
+    @unittest.skipIf(test.settings.TOPOLOGY_FROM_RESOURCE_SERVER, 'Test must be run from the Catalog Service Provider')
+    def test_server_returns_SYS_EXCEED_CONNECT_CNT_on_infinite_server_redirect_loop__issue_8529(self):
+        # This test requires that the Catalog Service Provider be identifiable by two names. This
+        # requirement is satisfied by the iRODS Testing Environment because setup_irods.py produces a
+        # server_config.json which initializes the "host" property to the container id while also allowing
+        # the same container to be identified (internally) as "irods-catalog-provider".
+
+        # Force the test to fail if it is not run from the provider.
+        hostname_alias = 'irods-catalog-provider'
+        with session.make_session_for_existing_admin(hostname_alias) as admin:
+            admin.assert_icommand(['imiscsvrinfo'], 'STDOUT', ['RCAT_ENABLED'])
+
+        # Force the test to fail if the provider cannot be identified by at least two different names.
+        if IrodsConfig().server_config['host'] == hostname_alias:
+            self.fail('[host] property in server_config.json cannot match [irods-catalog-provider]')
```

Comment:
This could be a `self.assertEqual`, *or*, if this is a precondition for the test, we have in other instances raised an `EnvironmentError` when a test's pre-conditions are not being met: https://github.com/irods/irods/blob/c1fc9f2c4835c51355ec4b277bd3a663e4f9e095/scripts/irods/test/test_pam_password_authentication.py#L79-L89

Reply:
Will update.

Reply:
Done.